modules/caddyhttp/fileserver/browse.go

```diff
@@ -111,13 +132,59 @@ func (fsrv *FileServer) serveBrowse(fileSystem fs.FS, root, dirPath string, w ht
 
 	acceptHeader := strings.ToLower(strings.Join(r.Header["Accept"], ","))
 
-	// write response as either JSON or HTML
-	if strings.Contains(acceptHeader, "application/json") {
+	// Next block of code makes the guarantee
+	w.Header().Set("Content-Type", acceptHeader)
+
+	// write response as one of the return types or HTML
+	returnType := (func() ReturnType {
+		if strings.Contains(acceptHeader, "/*") {
+			return fsrv.Browse.ReturnTypeRaw
+		}
+
+		if strings.Contains(acceptHeader, "application/json") {
+			return returnJSON
+		}
+
+		if strings.Contains(acceptHeader, "text/plain") {
+			return returnPlain
+		}
+
+		return returnTemplate
+	})()
+
+	if returnType == returnJSON {
```

Comment:
I find `switch` statements easier to read in these situations. Maybe let's try that?

Reply:
Sure, we can do that. I pushed a commit to change it.